### PR TITLE
UCT/TCP: Simplify TX/RX mpool PUT/GET using common macros

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Replace TCP code for put/get mpool elements by TX/RX mpool common PUT/GET macros

## Why ?

Simplify code and remove duplication

## How ?

Use common wrappers for TX/RX mpool PUT/GET

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11726/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 2 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 12-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/098aabee830a2e737da64c2b29823ed07afadcf0/_main) | :x: FAILURE
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/099f84a985ca323fe091be9722236bed8ada2eb8/hpc-arm-hwi-jenkins_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/597022bbe0a7ce4033472fb7321885791f34337c/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/c1492ac5a42d60d31551990b4c017c7f8e857ea6/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/f7389330a138e2b2834c4afa7fd7583769d71198/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/724e39b689c3a0db0c74df0f2cb01654117f7d25/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/bff9ae1f9193b4619bf4a6c2f9cdec4df61c1611/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/578dfe6224186e3e901b09a852957855eb9d96cf/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/d36768bbc89e6270386c1bea116915abb59cbb2e/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/9c485eeea96e17e12e8eb2875d2fe7ba2560bedc/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/1106c264565f13e0ec5b41110faa13d15e5f8552/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/f42894a0d4f5ab3427d99b62ce81328a6de2cab1/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/4ea38abf14269121b34ae5d720714b2d831415c1/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/a756bd12e009cac6b5bea9ebc152dd267b911dd8/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/45ee0819d18c2d9d342efd04ce1d3291f3e9391d/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/ddfa4e6e6769f27d5db22c62b25a7b79f5cc5896/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/d036338be380627146c55bc36c9245497bee5279/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/31c6fabc25b1f627f4fdfac8ae0ca3ee959dd751/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/842da5d5917507dcf0a62a6225120eb7faf76b2d/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/c1c4be17ab8b794ad682956175b99e7fe2c4d06d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/3fd7d0eb3be9495a2c057edca3abf588480b4efa/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/34fad3a7a461a6e49bbe774240dd68858b90a217/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/b90f0b1a05af7c62e796c03293600720dc2041f5/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/6d6ac195c06672f3cf2ad63e44430e16e870754a/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/ed72bebd99b1bdd314d866757949d9819aa18fd3/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/e1541b34e175b06674e0629834f7e787fc6c5f76/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/06c122d7c2509669bf72e01098a4afa511a8da6c/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/140c3541e5beb986f9c5fd54bc3c00919fa2a562/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/38207a3cdf24df8be39caef8e0448951/raw/4b62d81269fa82dd05666ae41c9bd38e3fa32e2c/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by dmitrygx** [comment]
bot:retest

File: src/uct/tcp/tcp_ep.c [hunk 1]
```diff
@@ -697,11 +697,8 @@ unsigned uct_tcp_ep_progress_rx(uct_tcp_ep_t *ep)
     if (!uct_tcp_ep_ctx_buf_need_progress(&ep->rx)) {
         ucs_assert(ep->rx.buf == NULL);
 
-        ep->rx.buf = ucs_mpool_get_inline(&iface->rx_mpool);
-        if (ucs_unlikely(ep->rx.buf == NULL)) {
-            ucs_warn("tcp_ep %p: unable to get a buffer from RX memory pool", ep);
-            return 0;
-        }
+        UCT_TL_IFACE_GET_RX_DESC(&iface->super, &iface->rx_mpool,
```
Comment (brminich) [inline src/uct/tcp/tcp_ep.c]: this macro makes `VALGRIND_MAKE_MEM_DEFINED(_desc, sizeof(*(_desc)));`
How would it work with `ep->(tx|rx).buf`, wich are `void*`?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11727/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 13-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/6ff9b6d34d705fb27019a82ec90a12f27909f2f9/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/52704f3ca875cb8012c413878ab69f4fc3112345/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/acf9ce66956f143f54222b9c85a4829ac6ba8c31/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/a12a74befd3c1a603c2799e0dd3f436b302074c4/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/f1b2b86765893bb9898c04b8eb8d77ccb3894038/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/08d675f97b6bf8ce2f9232dc231e7a4f051de8c0/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/7fe0484457065f247ddd992fe8beb764f9ae9a04/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/369f04ea09ac470b155686e67ab201546116efcb/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/9fa9a73f65e7502211bde537f6745624164fb2fb/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/cf32a1f1308224c42056e6863bd379c443802fe3/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/c0fcffb552686e191a0e4592fd0b5c47d1b55e80/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/ac01b1abf332d33262ba2570616a1da0d654160f/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/e2d02c734e720729c55c1d7d5a63b495e26882dd/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/669f420c09900185155ba0dd778b0094523a11d4/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/64d71cadf8f000a577ebedce45fb5bbb8ea9aa01/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/80d73bce5f3208f33888af1e23dd6b7492c4cf3f/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/a3239eebe3bc780235161964e53725d199df1feb/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/8319a3772d2971781968db32be2bf6938326014e/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/9abbab134d821ac9e0af3f7cc8f0828d93ecfa26/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/fa31a12d6f56270a09399932b05559d8fe3a299c/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/ea8f4fa3507205445664c9fe24d068eddc9f3e0f/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/8ce2e711695eb32672e281b56c9b63c0280612e3/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/672b1188bd7641bb990723a628ddc7627439ad75/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/53be6bce292e3028a85caf5b2962656e00db1049/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/02d6bf5d38ce00d51915e3b205a67eaed80dc854/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/08380fa9c9eeb26080fb7a161fff5f9e9689a1a4/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/babd530d9c639ac52c74f574a9ce2785bde811d6/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/1110fe9342b46c6dfd121750b033c5f21eb0df46/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb4e85318b5f96c740a955856874fb5c/raw/15da1adbc7c912da538c56f4a04292abfef1284f/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/uct/tcp/tcp_ep.c (hunk 1 above)
Comment (dmitrygx) [reply src/uct/tcp/tcp_ep.c]: good catch, I don't know how does it work, the same is in UGNI/UDT

it will be fixed shortly

**Comment by yosefe** [comment]
@dmitrygx is this still relevant?

**Comment by dmitrygx** [comment]
> @dmitrygx is this still relevant?

no, we can close this

**Pull request closed without merging**